PowerUp/app/src/main/java/powerup/systers/com/StartActivity.java

```diff
@@ -29,14 +29,25 @@ public void onClick(View v) {
 			}
 		});
 
+
 		ImageButton start = (ImageButton) findViewById(R.id.startButtonMain);
 		start.setOnClickListener(new View.OnClickListener() {
 			@Override
 			public void onClick(View v) {
-				Intent myIntent = new Intent(StartActivity.this, MapActivity.class);
-				startActivityForResult(myIntent, 0);
+				if (!previouslyStarted) {
```

Comment:
Make this if the other way round. Check for the positive condition and have the negative one in the else. It makes it more readable